float32 xdiff
float32 ydiff
float32 response
